File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/impl/metamodel/hashmap/aas/submodelelement/property/blob/Blob.java
package org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.submodelelement.property.blob;

import org.eclipse.basyx.aas.api.metamodel.aas.submodelelement.property.blob.IBlob;
import org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.submodelelement.DataElement;
import org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.submodelelement.property.Property;

/**
 * A blob element as defined in DAAS document <br/>
 * 
 * @author pschorn, schnicke
 *
 */
public class Blob extends DataElement implements IBlob{
	private static final long serialVersionUID = 1L;
	
	public static final String MIMETYPE="mimeType";

	/**
	 * Has to have a MimeType
	 * 
	 * @param value
	 *            the value of the BLOB instance of a blob data element
	 * @param mimeType
	 *            states which file extension the file has; Valid values are defined
	 *            in <i>RFC2046</i>, e.g. <i>image/jpg</i>
	 */
	public Blob(byte[] value, String mimeType) {
		super();

		put(Property.VALUE, value);
		put(MIMETYPE, mimeType);
	}

	@Override
	public void setValue(byte[] value) {
		put(Property.VALUE, value);
		
	}

	@Override
	public byte[] getValue() {
		return (byte[]) get(Property.VALUE);
	}

	@Override
	public void setMimeType(String mimeType) {
		put(Blob.MIMETYPE, mimeType);
		
	}

	@Override
	public String getMimeType() {
		return (String) get(Blob.MIMETYPE);
	}
}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/map/submodelelement/property/Blob.cpp
#include <BaSyx/submodel/map/submodelelement/property/Blob.h>

namespace basyx {
namespace submodel {

const std::string & Blob::getValue() const
{
	return map.getProperty(Blob::Path::Value).Get<std::string&>();
}

const std::string & Blob::getMimeType() const
{
	return map.getProperty(Blob::Path::MIMEType).Get<std::string&>();
}

void Blob::setValue(const std::string & bytes)
{
	this->map.insertKey(Blob::Path::Value, bytes);
}

void Blob::setValue(const char * c, std::size_t length)
{
// TODO: basyx::object for blob types
//	this->map.insertKey(Blob::Path::Value, basyx::object::make_blob)
}

void Blob::setMimeType(const std::string & mimeType)
{
	this->map.insertKey(IBlob::Path::MIMEType, mimeType, true);
}


}
};